### PR TITLE
feat: allow customization of trusting_period

### LINKED ISSUE #70: Allow customization of `trusting_period`

### PR DESCRIPTION
**Pull request by gjermundgaraba**
closes #70

### CODE REVIEW

File: operator/src/runners/fixtures/membership.rs [hunk 1]
```diff
@@ -66,8 +66,11 @@ pub async fn run(args: MembershipCmd) -> anyhow::Result<()> {
         .seconds
         .try_into()?;
 
+    // It is recommended that TrustingPeriod should be set as 2/3 of the UnbondingPeriod
+    let trusting_period = 2 * (unbonding_period / 3);
```
Comment (srdtrk) [inline operator/src/runners/fixtures/membership.rs:70]: Hm, don't you think we should allow setting of trusting period to any time less than or equal to the unbonding period? So basically adding a cli argument `--trusting-period` to genesis and maybe even fixtures cli. And if it is not provided, then use 2/3 of trusting period?

Comment (gjermundgaraba) [reply operator/src/runners/fixtures/membership.rs:70]: 🤦 you are of course right. Sorry, will add that xD

File: operator/src/runners/fixtures/membership.rs [hunk 2]
```diff
@@ -66,8 +66,14 @@ pub async fn run(args: MembershipCmd) -> anyhow::Result<()> {
         .seconds
         .try_into()?;
 
-    let trusted_client_state =
-        trusted_light_block.to_sol_client_state(args.trust_level.try_into()?, unbonding_period)?;
+    // Defaults to the recommended TrustingPeriod: 2/3 of the UnbondingPeriod
```
Comment (srdtrk) [inline operator/src/runners/fixtures/membership.rs:69]: Btw, where does this default value come from?

File: operator/src/cli/command.rs [hunk 3]
```diff
@@ -107,6 +110,10 @@ pub mod fixtures {
             help = "Trust level as a fraction, e.g. '2/3'",
         )]
         pub trust_level: super::TrustThreshold,
+
+        /// Trusting period. [default: 2/3 of unbonding period]
+        #[clap(long)]
+        pub trusting_period: Option<u32>,
```
Comment (srdtrk) [inline operator/src/cli/command.rs:116]: Since `trust_level` + `trusting_period` are often bundled together (i.e. if one is present in the cli args, then so is the other), it is possible to move them both to a single struct, called something like `trust_options`. And then use flatten in the actual command struct so that the user can just input `--trust-level` and `--trust-period`. This reduces code duplication. Do you wanna do this refactor here or just merge this and then we can open a separate PR for the refactor?

File: justfile [hunk 4]
```diff
@@ -45,10 +45,10 @@ fixtures prover:
...
+    "{{ if prover == "network" { "sleep 30 &&" } else { "" } }} RUST_LOG=info SP1_PROVER={{prover}} TENDERMINT_RPC_URL='https://rpc.celestia-mocha.com/' ./target/release/operator fixtures membership --key-paths 'clients/07-tendermint-0/clientState' --trusted-block 2330000 --trusting-period 1209600 -o 'contracts/fixtures/{{ if prover == "mock" { "mock_" } else { "" } }}verify_membership_fixture.json'" \
+    "{{ if prover == "network" { "sleep 45 &&" } else { "" } }} RUST_LOG=info SP1_PROVER={{prover}} TENDERMINT_RPC_URL='https://rpc.celestia-mocha.com/' ./target/release/operator fixtures membership --key-paths clients/07-tendermint-0/clientState,clients/07-tendermint-001/clientState --trusted-block 2330000 --trusting-period 1209600 -o 'contracts/fixtures/{{ if prover == "mock" { "mock_" } else { "" } }}memberships_fixture.json'"
```
Comment (srdtrk) [inline justfile:51]: Why did you add this argument if it is optional?

File: e2e/interchaintestv8/sp1_ics07_test.go [hunk 5]
```diff
@@ -110,10 +110,12 @@ func (s *SP1ICS07TendermintTestSuite) TestDeploy() {
 		stakingParams, err := simd.StakingQueryParams(ctx)
 		s.Require().NoError(err)
 
+		expectedTrustingPeriod := (stakingParams.UnbondingTime.Seconds() / 3) * 2
```
Comment (srdtrk) [inline e2e/interchaintestv8/sp1_ics07_test.go:113]: Maybe we should test a custom value in the e2e tests. Similar to how I pass a custom value for trust level.

**Review by srdtrk — APPROVED** [review]
Looks good to me! Thanks. I left a couple questions. Can merge this once we resolve those

File: operator/src/runners/fixtures/membership.rs (hunk 2 above)
Comment (gjermundgaraba) [reply operator/src/runners/fixtures/membership.rs:69]: I've seen it used other places, but I found it when reading up on the value here: https://tutorials.cosmos.network/academy/3-ibc/4-clients.html

The CometBFT docs only says "This value should be significantly smaller than the unbonding period." 🤷

File: justfile (hunk 4 above)
Comment (gjermundgaraba) [reply justfile:51]: Good question :D Mostly just to have it available as an example in the just file, but it doesn't make a lot sense. I will remove it :)

File: e2e/interchaintestv8/sp1_ics07_test.go (hunk 5 above)
Comment (gjermundgaraba) [reply e2e/interchaintestv8/sp1_ics07_test.go:113]: Good idea! Will do that.
